### ISSUE TITLE
Negative Fis

### ISSUE THREAD

**Issue by nickbrazeau** [nickbrazeau/discent#10]
Able to get negative values of Fi with code below (my computer seed) 

- there is nothing mathematically stopping Fi from being negative. conceptually does it make sense? think of relatedness as bounded but this is a inbreeding coeff 

```
library(discent)
set.seed(48)
#......................
# sim
#......................
distmat <- matrix(c(0,100,200,100,0,300,200,300,0), nrow = 3, ncol = 3)
input <-  discent::sim_IBDIBD(demesize = c(3,5,5),
                              distmat = distmat,
                              rate = 1e-2,
                              Ft = 0.3)
input <- input %>%
  dplyr::filter(locat1 != locat2)

#......................
# run discent
#......................
our_start_params <- rep(0.2, 3)
names(our_start_params) <- 1:3
our_start_params <- c(our_start_params, "m" = 1e-5)
ret <- discent::deme_inbreeding_spcoef(K_gendist_geodist = input,
                                       start_params = our_start_params,
                                       f_learningrate = 1e-5,
                                       m_learningrate = 1e-10,
                                       momentum = 0.9,
                                       steps = 1e4,
                                       report_progress = T,
                                       return_verbose = T)
ret$Final_m
ret$Final_Fis

# cost
plot(ret$cost)
summary(ret$cost)
ret$cost[length(ret$cost)]
plot(ret$cost[(length(ret$cost)-500):length(ret$cost)])

# runs
plot(ret$m_run)
plot(ret$fi_run[,1])
plot(ret$fi_run[,3])
# updates
plot(ret$m_update)
plot(ret$fi_update[,1])
plot(ret$fi_update[,3])
```

**Comment by nickbrazeau** [comment]
I can do a logit (and expit) transformation of the data which bounds and "solves" this issue...but is that conceptually correct? 

- Mathematically, this transformation should be valid and does not affect the gradient descent as it happens "upstream" of the Y and P declaration 
- _However, interestingly_, the logit transformation appears to increase instability in the model

**Comment by nickbrazeau** [comment]
logit and expit solves this issue. is fine
